cpmpy/solvers/gurobi.py

```diff
@@ -0,0 +1,376 @@
+    def minimize(self, expr):
+        """
+            Minimize the given objective function
+
+            `minimize()` can be called multiple times, only the last one is used
+
+            (technical side note: any constraints created during conversion of the objective
+            are premanently posted to the solver)
+        """
+        from gurobipy import GRB
+
+        # make objective function non-nested
+        (flat_obj, flat_cons) = (flatten_objective(expr))
+        self += linearize_constraint(flat_cons)  # add potentially created constraints
+
+        # make objective function or variable and post
+        obj = self._make_numexpr(flat_obj)
+        self.gbi_model.setObjective(obj, sense=GRB.MINIMIZE)
+
+    def maximize(self, expr):
+        """
+            Maximize the given objective function
+
+            `maximize()` can be called multiple times, only the last one is used
+
+            (technical side note: any constraints created during conversion of the objective
+            are premanently posted to the solver)
+        """
+        from gurobipy import GRB
+
+        # make objective function non-nested
+        (flat_obj, flat_cons) = (flatten_objective(expr))
+        self += linearize_constraint(flat_cons)  # add potentially created constraints
+
+        # make objective function or variable and post
+        obj = self._make_numexpr(flat_obj)
+        self.gbi_model.setObjective(obj, sense=GRB.MAXIMIZE)
+
+    def _make_numexpr(self, cpm_expr):
+        """
+            Turns a numeric CPMpy 'flat' expression into a solver-specific
+            numeric expression
+
+            Used especially to post an expression as objective function
+        """
+        import gurobipy as gp
+
+        if is_num(cpm_expr):
+            return cpm_expr
+
+        # decision variables, check in varmap
+        if isinstance(cpm_expr, _NumVarImpl):  # _BoolVarImpl is subclass of _NumVarImpl
+            return self.solver_var(cpm_expr)
+
+        # sum
+        if cpm_expr.name == "sum":
+            return gp.quicksum([self.solver_var(var) for var in cpm_expr.args])
+
+        if cpm_expr.name == "wsum":
+            weights, cpm_vars = cpm_expr.args
+            gbi_vars = [self.solver_var(var) for var in cpm_vars]
+            return gp.LinExpr(weights, gbi_vars)
+
+        raise NotImplementedError("gurobi: Not a know supported numexpr {}".format(cpm_expr))
+
+    def _post_constraint(self, cpm_expr):
+        """
+            Post a primitive CPMpy constraint to the native solver API
+
+            What 'primitive' means depends on the solver capabilities,
+            more specifically on the transformations applied in `__add__()`
+
+            Solvers do not need to support all constraints.
+        """
+
+        # Base case: Boolean variable
+        if isinstance(cpm_expr, _BoolVarImpl):
+            return self.gbi_model.addLConstr(self.solver_var(cpm_expr), ">", 1)
```

Comment:
I don't understand that you write ">" for sense (2nd argument), sense should be:
   sense: Sense for the new constraint (GRB.LESS_EQUAL, GRB.EQUAL, or GRB.GREATER_EQUAL).
so ">" is confusing to me

Reply:
maybe this should be done in linearize actually? just like you translate an or?

Reply:
Would indeed make more sense to put in the linearization lib. I will add it there